### ISSUE TITLE
🐛 [firebase_analytics] Measuring in-app purchases manually required by StoreKit 2

### ISSUE THREAD

**Issue by ninjz** [invertase/react-native-firebase#8077]
Apparently when supporting StoreKit 2 on iOS, in-app purchases cannot be measured automatically by the analytics SDK anymore. https://firebase.google.com/docs/analytics/measure-in-app-purchases#implementation

I can't seem to find a method for manually tracking IAP events. Are there plans on supporting manually logging this or is this already supported somewhere?

**Comment by mikehardy** [comment]
Apologies, I do not know - this might be something to ask directly on stackoverflow tagging it appropriately (firebase team offers support there) or through your normal firebase support channels as opposed to this repo

Whatever they suggest is what we could aim for here

**Comment by ninjz** [comment]
I think it’s quite relevant as a feature request to this repo seeing as this would require additional method calls to the native SDKs to support this. 

There are already calls to log non IAP purchases. This would be an extension as auto-logging of IAP events will not work with the deprecation of StoreKit 1. 

I believe this would be required by anyone interacting Purchase SDKs and need to send IAP purchase events back to Firebase analytics when StoreKit 2 becomes mandatory.

**Comment by mikehardy** [comment]
Totally agreed, the problem is I'm unsure what the native method calls *are* in the SDK and/or how firebase-js-sdk (our reference API) does it - do you have a link to their documentation and APIs / types for this perhaps

If not:

> This might be something to ask directly on stackoverflow tagging it appropriately (firebase team offers support there) or through your normal firebase support channels as opposed to this repo

**Comment by LectricAvenue** [comment]
@ninjz curious if you made any headway with this. I am about to start looking into this for a solution myself and figured I would check in here first.

**Comment by gbesler** [comment]
Any solutions on this? It has been over a year! Firebase is not reporting in app purchases with Storekit 2.

**Comment by mikehardy** [comment]
@gbesler - nope, looks like the free work fairy hasn't shown up yet ;-). I left some ideas that would move this forward, and apparently it isn't important enough for anyone to invest time in yet. If it is important to you please let me know what the results are of investigating for relevant APIs in the firebase upstream SDKs and/or stackoverflow

**Comment by gbesler** [comment]
Oh no :( Well, we use RevenueCat, and it is important to us to report native in app purchase events. But since RevenueCat does not provide Transactions we cannot use the way Firebase recommends for StoreKit 2. Firebase told they will deliver this in 2024 Q4, still nothing.

Do you have any idea how to solve this myself?

**Comment by mikehardy** [comment]
@gbesler with apologies no, I don't have any ideas. I see you participated (briefly! they closed the issue...) upstream in firebase-ios-sdk - that along with contacting Firebase Support appears to be the avenue based on upstream conversation. We can't really do anything until they fix it, then...I assume we would just update the firebase-ios-sdk version here and it would Just Work

**Comment by mikehardy** [comment]
Oh wait look at this! firebase-ios-sdk 11.7.0 has some mention of it in release notes:
https://firebase.google.com/support/release-notes/ios#analytics

You might like to try result of this PR once merged + released

- #8232

**Comment by gbesler** [comment]
Oh wow. I guess they released it. We will try it ASAP. Many thanks @mikehardy 
I hope it just works.

**Comment by mikehardy** [comment]
@gbesler I've got some problem with the iOS 18.2 SDK not being installed on our CI runner even though Xcode 16.2 is selected 🤷  so I'm not able to merge immediately and I can't say how long it will take.

However, if you're not aware, you can quickly override the firebase-ios-sdk version in your Podfile for local integration to see how it goes - just remember to remove the override later if you use it: https://rnfirebase.io/#ios

If you're using Expo I'm not aware of a way to override like that unfortunately

**Comment by gbesler** [comment]
Hey @mikehardy , We have implemented 11.7.0 but unfortunately no change. It is still not working. We don't see 90% of the events on the panel same as before.

**Comment by mikehardy** [comment]
Sorry to hear it. Definitely post findings upstream if you haven't already, so they are aware that we've checked the most recent work in the area

**Comment by gbesler** [comment]
Any improvements on your side @mikehardy ?

**Comment by mikehardy** [comment]
Nope, not something I'm actively pursuing as it does not appear there is anything we can do to affect it in react-native-firebase. Appears to be upstream. So my main task (which takes a non-trivial amount of time!) is to make sure react-native-firebase is always up to date and releasable (which it is, just did a release a couple minutes ago after updating our e2e to new arch etc. Keeping it modern...).
